### PR TITLE
annotate new_type and exec structs with must_use

### PR DESCRIPTION
**Pull request by hdhoang**
Thank you for your detailed instructions. I have implemented them here.

### CODE REVIEW

**Comment by hdhoang** [comment]
Ah, I think there might be an issue with recompiled artifacts. I have hit `multiple candidates for crate github_rs [E0464]` while working on this, and cleaning the `target` dir fixed it. If we can't make it work reliably, you can disable the tests.

**Comment by mgattozzi** [comment]
Hey @hdhoang thanks for the PR! I'm heading to a wedding weekend today so I might not get to this till tomorrow (since it's a free day) or until this weekend, but I wanted to let you know I saw this! I will take a look at it as soon as I can. I'll check out that behavior while reviewing your PR!

**Comment by hdhoang** [comment]
I'm trying to move the test to UI integration instead of compile-fail, as per compiletest's recommendation. This should reflect the intention better.

**Comment by mgattozzi** [comment]
I like that. Ping me when this is ready for review @hdhoang!

**Comment by hdhoang** [comment]
This is ready for review! Thanks!

**Comment by hdhoang** [comment]
The test is gated behind `--feature compiletest` now.

**Comment by hdhoang** [comment]
I have pulled the compile tests out into a separate workspace like diesel's. As a bonus, this avoids E0464 for our usecase, and also restores `cargo +stable test`.

The last commit also corrects "[workspace**s**]" typo in `Cargo.toml`.

**Comment by mgattozzi** [comment]
Hey @hdhoang want to let you know that I see this, I'm just a bit busy today and visitng my brother in Texas tomorrow so I might not get to this till I find some downtime and internet acess but I hope too soon!

**Comment by mgattozzi** [comment]
Hey @hdhoang if you rebase this I can try to get this in asap :D

**Comment by hdhoang** [comment]
I have rebased onto 0.6.1

**Comment by hdhoang** [comment]
I have rebased again. I also joined `compile_tests` into the parent workspace, as `compiletest-rs 0.3.4` has fixed the "check then test" issue.

The main project is still stable-compatible, `cargo +nightly` is only required for running tests within the `compile_tests` directory.

File: src/gists/delete.rs [hunk 1]
```diff
@@ -4,11 +4,8 @@ use client::DeleteQueryBuilder;
 
 new_type!(
     Gists
-    GistsId
```
Comment (Dylan-DPC) [inline src/gists/delete.rs]: why are these removed? cc @mgattozzi

Comment (hdhoang) [reply src/gists/delete.rs]: The `pub struct $i<'g>` type declaration is duplicated into `exec!`, so that the two declarations can have different `#[must_use]` annotations explaining their usage. Thus, types which are declared in `exec!` must now be removed from `new_type!`.

The original vision was laid out in #87.

Comment (Dylan-DPC) [reply src/gists/delete.rs]: Ok thanks. This looks good. Merging it :)
